src/main/java/org/wso2/securevault/secret/repository/VaultSecretRepositoryProvider.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.securevault.secret.repository;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.securevault.SecureVaultException;
+import org.wso2.securevault.commons.MiscellaneousUtil;
+import org.wso2.securevault.keystore.IdentityKeyStoreWrapper;
+import org.wso2.securevault.keystore.TrustKeyStoreWrapper;
+import org.wso2.securevault.secret.SecretRepository;
+import org.wso2.securevault.secret.SecretRepositoryProvider;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+public class VaultSecretRepositoryProvider implements SecretRepositoryProvider {
+
+    private static Log log = LogFactory.getLog(VaultSecretRepositoryProvider.class);
+
+    /* Property String for secretProviders.*/
```

Comment:
The comment style seems odd. Better to stick to convensions

Reply:
Changed